B2LScrape - got an antibiotic resistance problem?
=================================================

Scrape data on ÃŸ-Lactamase Classification and Amino Acid Sequences for TEM, 
SHV and OXA Extended-Spectrum and Inhibitor Resistant Enzymes

From: http://www.lahey.org/Studies/

Thanks to George Jacoby and Karen Bush for making such as useful resource
available.

**No scraping yet.** Tables have been pulled manually. See: tables/


Requirements
------------

Python 2.7 & biopython.


Usage
-----

Something like::

    $ ./B2LScrape.py -h
    
    usage: B2LScrape.py [-h] [-v] [--version] csv_file

    B2LScrape v0.1 - Fetch Beta-Lactamase sequences from the NCBI

    positional arguments:
      csv_file       Full path to the input csv

    optional arguments:
      -h, --help     show this help message and exit
      -v, --verbose  verbose output
      --version      show program's version number and exit

    Licence: ECL 2.0 by Mitchell Stanton-Cook <m.stantoncook@gmail.com>


CSV Table format
----------------

In CSV. Formatted with elements: id, accession (if available)

Example::

    ACC-1,AJ133121
    ACC-2,AF180952
    ACC-3,AF180958
    ACC-4,EF504260
    ACC-5,


Tutorial/Quickstart
-------------------

Firstly, as the NCBI accessions are sometimes to entire plasmids, B2LScrape
has to be fairly interactive. Sorry, but would you like to do this manually in
your web browser?

Grab the BEL-type Beta-Lactamases::

    $ ./B2LScrape.py tables/BEL.csv 
    
    Working with BEL-1: 
    (4)*** aacA4. Aac6'-Ib. aminoglycoside acetyltransferase
    (8)*** bla(BEL1). Bel-1. extended-spectrum beta-lactamase
    (12)*** smr2. Smr2. small drug resistance protein
    (16)*** aadA5. AadA5. adenylytransferase
    Index of selection: 8
    Working with BEL-2: 
    (1)***  . extended-spectrum beta-lactamase BEL-2.  
    Index of selection: 1
    Working with BEL-3: 
    (2)*** blaBEL-3. BEL-3. BEL-type ESBL; BEL variant
    Index of selection: 2

Gives::

    $ cat tables/BEL.fa 
    
    >DQ089809, BEL-1, aadA5. AadA5. adenylytransferase, Unknown sp. [BEL]
    ATGAAACTGCTGCTCTACCCGTTATTGCTGTTCCTTGTCATTCCAGCCTTTGCCCAGGCGGACTTTGAACATGCCATTTC
    AGATCTTGAGGCGCACAATCAAGCCAAGATCGGAGTGGCCCTAGTTAGTGAAAATGGCAACCTGATTCAAGGGTATCGTG
    CGAATGAAAGGTTCGCGATGTGCTCAACTTTCAAGTTGCCGTTGGCCGCTCTTGTTCTGAGTCGCATTGACGCTGGGGAA
    GAGAATCCTGAGCGCAAGCTTCATTACGATTCCGCGTTCCTTGAAGAGTACGCCCCAGCCGCAAAACGGTATGTGGCAAC
    TGGATATATGACTGTAACTGAGGCAATTCAATCCGCCCTCCAACTCAGCGACAATGCCGCAGCTAACCTGCTGTTAAAAG
    AGGTTGGCGGCCCACCTTTATTGACAAAGTATTTCCGTAGCCTGGGTGATAAAGTAAGTCGCCTTGATCGTATTGAACCG
    ACTTTGAACACCAATACGCCCGGCGATGAAAGAGATACAACAACGCCCATGTCCATGGCACAGACTGTGTCAAAGCTGAT
    TTTTGGAGACACGTTGACATATAAATCCAAGGGGCAGCTAAGGCGATTACTCATCGGCAATCAGACCGGGGACAAAACCA
    TTCGAGCTGGCTTGCCTGATTCATGGGTAACGGGTGACAAGACAGGCTCGTGTGCGAATGGCGGCCGTAACGATGTGGCG
    TTTTTTATAACCACTGCCGGAAAAAAATATGTTCTTTCTGTATATACCAATGCACCTGAATTGCAAGGCGAGGAAAGGGC
    GTTATTAATTGCTTCTGTAGCAAAGTTAGCACGTCAATATGTTGTTCACTGA
    >FJ666063, BEL-2, . extended-spectrum beta-lactamase BEL-2., Unknown sp. [BEL]
    ATGAAACTGCTGCTCTACCCGTTATTGCTGTTCCTTGTCATTCCAGCCTTTGCCCAGGCGGACTTTGAACATGCCATTTC
    AGATCTTGAGGCGCACAATCAAGCCAAGATCGGAGTGGCCCTAGTTAGTGAAAATGGCAACCTGATTCAAGGGTATCGTG
    CGAATGAAAGGTTCGCGATGTGCTCAACTTTCAAGTTGCCGTTGGCCGCTCTTGTTCTGAGTCGCATTGACGCTGGGGAA
    GAGAATCCTGAGCGCAAGCTTCATTACGATTCCGCGTTCCTTGAAGAGTACGCCCCAGCCGCAAAACGGTATGTGGCAAC
    TGGATATATGACTGTAACTGAGGCAATTCAATCCGCCCTCCAACTCAGCGACAATGCCGCAGCTAACCTGCTGTTAAAAG
    AGGTTGGCGGCCCACCTTTATTGACAAAGTATTTCCGTAGCCTGGGTGATAAAGTAAGTCGCTTTGATCGTATTGAACCG
    ACTTTGAACACCAATACGCCCGGCGATGAAAGAGATACAACAACGCCCATGTCCATGGCACAGACTGTGTCAAAGCTGAT
    TTTTGGAGACACGTTGACATATAAATCCAAGGGGCAGCTAAGGCGATTACTCATCGGCAATCAGACCGGGGACAAAACCA
    TTCGAGCTGGCTTGCCTGATTCATGGGTAACGGGTGACAAGACAGGCTCGTGTGCGAATGGCGGCCGTAACGATGTGGCG
    TTTTTTATAACCACTGCCGGAAAAAAATATGTTCTTTCTGTATATACCAATGCACCTGAATTGCAAGGCGAGGAAAGGGC
    GTTATTAATTGCTTCTGTAGCAAAGTTAGCACGTCAATATGTTGTTCACTGA
    >GQ202694, BEL-3, blaBEL-3. BEL-3. BEL-type ESBL; BEL variant, Unknown sp. [BEL]
    ATGAAACTGCTGCTCTACCCGTTATTGCTGTTCCTTGTCATTCCAGCCTTTGCCCAGGCGGACTTTGAACATGCCATTTC
    AGATCTTGAGGCGCACAATCAAGCCAAGATCGGAGTGGCCCTAGTTAGTGAAAATGGCAACCTGATTCAAGGGTATCGTG
    CGAATGAAAGGTTCGCGATGTGCTCAACTTTCAAGTTGCCGTTGGCCGCTCTTGTTCTGAGTCGCATTGACGCTGGGGAA
    GAGAATCCTGAGCGCAAGCTTCATTACGATTCCGCGTTCCTTGAAGAGTACGCCCCAGCCGCAAAACGGTATGTGGCAAC
    TGGATATATGACTGTAACTGAGGCAATTCAATCCGCCCTCCAACTCAGCGACAATGCCGCAGCTAACCTGCTGTTAAAAG
    AGGTTGGCGGCCCACCTTTATTGACAAAGTATTTCCGTAGCCTGGGTGATAAAGTAAGTCGCCTTGATCGTATTGAATCG
    ACTTTGAACACCAATACGCCCGGCGATGAAAGAGATACAACAACGCCCATGTCCATGGCACAGACTGTGTCAAAGCTGAT
    TTTTGGAGACACGTTGACATATAAATCCAAGGGGCAGCTAAGGCGATTACTCATCGGCAATCAGACCGGGGACAAAACCA
    TTCGAGCTGGCTTGCCTGATTCATGGGTAACGGGTGACAAGACAGGCTCGTGTGCGAATGGCGGCCGTAACGATGTGGCG
    TTTTTTATAACCACTGCCGGAAAAAAATATGTTCTTTCTGTATATACCAATGCACCTGAATTGCAAGGCGAGGAAAGGGC
    GTTATTAATTGCTTCTGTAGCAAAGTTAGCACGTCAATATGTTGTTCACTGA

I loaded these up into JalView and their is an allele change at positions 462 
(BEL-2) and 478 (BEL-3).

.. image:: https://raw.github.com/mscook/B2LScrape/master/BEL-screenshot.png
    :alt: BEL Screenshot
    :align: center


Whats with the fasta header formatting?
---------------------------------------

Is formatted in SeqFindR format. See http://github.com/mscook/SeqFindR

